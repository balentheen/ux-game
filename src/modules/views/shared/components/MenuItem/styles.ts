import { StyleSheet, Dimensions } from 'react-native';

import { STYLE } from '../../../../../constants';

export default StyleSheet.create({
  container: {
    width: Dimensions.get('window').width * 0.85,
    aspectRatio: 1,
    paddingVertical: 15,
    paddingHorizontal: 10,
    borderRadius: 4,
    backgroundColor: STYLE.COLOR.WHITE,
  },
  questionContainer: {
    width: '100%',
    height: '100%',
    backgroundColor: STYLE.COLOR.WHITE,
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: 5,
    shadowColor: STYLE.COLOR.BLACK,
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.2,
    shadowRadius: 5,
    elevation: 3,
    padding: 15
  },
  category: {
    position: 'absolute',
    bottom: 15,
    left: 15,
    fontSize: 9,
    fontWeight: STYLE.FONT.WEIGHTS.MEDIUM,
    color: STYLE.COLOR.PRIMARY,
    alignSelf: 'flex-end'
  },
  question: {
    marginTop: 10,
    fontSize: 60,
    fontFamily: 'montserrat-semibold',
    color: STYLE.COLOR.BLACK
  },
  section: {
    fontSize: 15,
    fontFamily: 'montserrat-regular',
    color: STYLE.COLOR.DIM_GRAY,
    textAlign: 'center'
  },
  trigger: {
    width: 16,
    height: 16,
    justifyContent: 'center',
    alignItems: 'flex-end'
  },
  bar: {
    width: 3,
    height: 16,
    left: 7,
    position: 'absolute',
    backgroundColor: STYLE.COLOR.CURIOUS_BLUE
  },
  horizontalBar: {
    transform: [{ rotate: '90deg' }]
  }
});
