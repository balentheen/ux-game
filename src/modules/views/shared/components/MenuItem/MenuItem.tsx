import React from 'react';
import { Text, View, TouchableOpacity, Image, ImageSourcePropType } from 'react-native';

import styles from './styles';

export interface IMenuItemProps {
  icon: ImageSourcePropType;
  title: string;
  description: string;
  onPress?: () => void;
}

export default class MenuItem extends React.PureComponent<IMenuItemProps, {}> {

  public render() {
    const { title, description, icon, onPress } = this.props;
    return (
      <View style={styles.container}>
        <TouchableOpacity onPress={onPress} style={styles.questionContainer}>
          <Image source={icon} />
          <Text style={styles.question}>{title}</Text>
          <Text style={styles.section}>{description}</Text>
        </TouchableOpacity>
      </View>
    );
  }
}
