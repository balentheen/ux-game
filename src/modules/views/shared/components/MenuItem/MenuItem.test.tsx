import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import MenuItem, { IMenuItemProps } from './MenuItem';

describe('MenuItem', () => {
  let props: IMenuItemProps;
  beforeEach(() => {
    props = {
      icon: 1,
      title: 'title',
      description: 'desc',
      onPress: jest.fn()
    };
  });

  it('renders without crashing', () => {
    const rendered = renderer.create(<MenuItem {...props} />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
