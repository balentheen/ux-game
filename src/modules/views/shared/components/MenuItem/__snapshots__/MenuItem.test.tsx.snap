// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FaqQuestion renders without crashing 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFF",
      "borderRadius": 4,
      "height": 147,
      "marginVertical": 5,
      "padding": 15,
      "width": 168,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#FFF",
        "borderRadius": 5,
        "elevation": 3,
        "height": "100%",
        "opacity": 1,
        "padding": 15,
        "shadowColor": "#1E272E",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0.2,
        "shadowRadius": 5,
        "width": "100%",
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#676767",
          "fontSize": 12,
          "fontWeight": "400",
        }
      }
    >
      Basics
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#1E272E",
          "fontSize": 13,
          "fontWeight": "600",
          "marginTop": 10,
        }
      }
    >
      Frequently asked question?
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "alignSelf": "flex-end",
          "bottom": 15,
          "color": "#0076FF",
          "fontSize": 9,
          "fontWeight": "600",
          "left": 15,
          "position": "absolute",
        }
      }
    />
  </View>
</View>
`;
